### PR TITLE
Use the Button component from graph-paper

### PR DESCRIPTION
**Pull request by openjck**

### CODE REVIEW

**Comment by openjck** [comment]
If this Rollup config looks good, I'll recommend it in the graph-paper README.

File: src/components/controls/BodyControl.svelte [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <script>
 import { createEventDispatcher } from 'svelte';
-import Button from 'udgl/Button.svelte';
+import Button from '@graph-paper/button';
```
Comment (hamilton) [inline src/components/controls/BodyControl.svelte]:
```suggestion
import { Button, ButtonGroup } from '@graph-paper/button';
```

File: src/components/controls/BodyControl.svelte [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 import ButtonGroup from 'udgl/ButtonGroup.svelte';
```
Comment (hamilton) [inline src/components/controls/BodyControl.svelte]:
```suggestion
```

File: src/components/controls/Pagination.svelte [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <script>
 import { createEventDispatcher } from 'svelte';
-import Button from 'udgl/Button.svelte';
+import Button from '@graph-paper/button';
```
Comment (hamilton) [inline src/components/controls/Pagination.svelte]:
```suggestion
import { Button, ButtonGroup } from '@graph-paper/button';
```

File: src/components/controls/Pagination.svelte [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
 import ButtonGroup from 'udgl/ButtonGroup.svelte';
```
Comment (hamilton) [inline src/components/controls/Pagination.svelte]:
```suggestion
```

File: stories/data-graphics/elements/Point01.svelte [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <script>
 import { fly, fade } from 'svelte/transition';
 
-import Button from 'udgl/Button.svelte';
+import Button from '@graph-paper/button';
```
Comment (hamilton) [inline stories/data-graphics/elements/Point01.svelte]:
```suggestion
import { Button, ButtonGroup } from '@graph-paper/button';
```

File: stories/data-graphics/elements/Point01.svelte [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
 import ButtonGroup from 'udgl/ButtonGroup.svelte';
```
Comment (hamilton) [inline stories/data-graphics/elements/Point01.svelte]:
```suggestion
```

File: stories/data-graphics/mozilla/ReleaseHealthDashboard.svelte [hunk 7]
```diff
@@ -8,7 +8,7 @@ import Axis from 'udgl/data-graphics/guides/Axis.svelte';
 import AxisLabel from 'udgl/data-graphics/guides/AxisLabel.svelte';
 import AxisLine from 'udgl/data-graphics/guides/AxisLine.svelte';
 import AxisTick from 'udgl/data-graphics/guides/AxisTick.svelte';
-import Button from 'udgl/Button.svelte';
+import Button from '@graph-paper/button';
```
Comment (hamilton) [inline stories/data-graphics/mozilla/ReleaseHealthDashboard.svelte]:
```suggestion
import { Button, ButtonGroup } from '@graph-paper/button';
```

File: stories/data-graphics/mozilla/ReleaseHealthDashboard.svelte [hunk 8]
```diff
@@ -8,7 +8,7 @@ import Axis from 'udgl/data-graphics/guides/Axis.svelte';
...
 import ButtonGroup from 'udgl/ButtonGroup.svelte';
```
Comment (hamilton) [inline stories/data-graphics/mozilla/ReleaseHealthDashboard.svelte]:
```suggestion
```

File: stories/layout/BigDrawerBody.svelte [hunk 9]
```diff
@@ -5,7 +5,7 @@ import Content from 'udgl/layout/Content.svelte';
 import ContentBody from 'udgl/layout/ContentBody.svelte';
 import ContentHeader from 'udgl/layout/ContentHeader.svelte';
 import ButtonGroup from 'udgl/ButtonGroup.svelte';
-import Button from 'udgl/Button.svelte';
+import Button from '@graph-paper/button';
```
Comment (hamilton) [inline stories/layout/BigDrawerBody.svelte]:
```suggestion
import { Button, ButtonGroup } from '@graph-paper/button';
```

File: stories/layout/BigDrawerBody.svelte [hunk 10]
```diff
@@ -5,7 +5,7 @@ import Content from 'udgl/layout/Content.svelte';
 import ContentBody from 'udgl/layout/ContentBody.svelte';
 import ContentHeader from 'udgl/layout/ContentHeader.svelte';
 import ButtonGroup from 'udgl/ButtonGroup.svelte';
```
Comment (hamilton) [inline stories/layout/BigDrawerBody.svelte]:
```suggestion
```

File: stories/layout/CenteredBody.svelte [hunk 11]
```diff
@@ -6,7 +6,7 @@ import ContentHeader from 'udgl/layout/ContentHeader.svelte';
 import ContentBody from 'udgl/layout/ContentBody.svelte';
 import ContentFooter from 'udgl/layout/ContentFooter.svelte';
 
-import Button from 'udgl/Button.svelte';
+import Button from '@graph-paper/button';
```
Comment (hamilton) [inline stories/layout/CenteredBody.svelte]:
```suggestion
import { Button, ButtonGroup } from '@graph-paper/button';
```

File: stories/layout/CenteredBody.svelte [hunk 12]
```diff
@@ -6,7 +6,7 @@ import ContentHeader from 'udgl/layout/ContentHeader.svelte';
...
 import ButtonGroup from 'udgl/ButtonGroup.svelte';
```
Comment (hamilton) [inline stories/layout/CenteredBody.svelte]:
```suggestion
```

File: stories/ux-components/ButtonGroup.svelte [hunk 13]
```diff
@@ -1,5 +1,5 @@
 <script>
-import Button from 'udgl/Button.svelte';
+import Button from '@graph-paper/button';
```
Comment (hamilton) [inline stories/ux-components/ButtonGroup.svelte]:
```suggestion
import { Button, ButtonGroup } from '@graph-paper/button';
```

File: stories/ux-components/ButtonGroup.svelte [hunk 14]
```diff
@@ -1,5 +1,5 @@
...
 import ButtonGroup from 'udgl/ButtonGroup.svelte';
```
Comment (hamilton) [inline stories/ux-components/ButtonGroup.svelte]:
```suggestion
```

File: rollup.config.js [hunk 15]
```diff
@@ -27,6 +28,11 @@ export default {
       css: (css) => {
         css.write('public/static/bundle.css');
       },
+      include: ['src/**/*.svelte', 'node_modules/@graph-paper/**/*.svelte'],
```
Comment (hamilton) [inline rollup.config.js]: Why is this line required? Shouldn't this work as-is?

**Review by hamilton — CHANGES_REQUESTED** [review]
Nice to see this PR get filed – thanks @openjck. I went through and added suggestions around importing `ButtonGroup` as well. Those should be easy to fix from the UI.

- is it necessary to add those `include` statements in the svelte rollup plugin? I would think that svelte would take care of all that separately by default.
- We will probably need to focus a small amount of attention on the button component in graph-paper, so I suspect the props API will change. We can revisit this later; these types of change are very easy to make in a repository down the line.

File: rollup.config.js (hunk 15 above)
Comment (openjck) [reply rollup.config.js]: Note to self: I was pretty sure that it was, but this component library doesn't require this:

https://github.com/hperrin/smui-example-rollup/blob/master/rollup.config.js

**Comment by openjck** [comment]
Updated

File: rollup.config.js (hunk 15 above)
Comment (openjck) [reply rollup.config.js]: You're right. I was playing around with a bunch of different settings and mistakenly thought this was one of the changes that was relevant, but it isn't. The PostCSS stuff is still needed, though.

**Comment by openjck** [comment]
Updated and ready for re-review! 😃

File: rollup.config.js [hunk 16]
```diff
@@ -5,6 +5,7 @@ import livereload from 'rollup-plugin-livereload';
 import { terser } from 'rollup-plugin-terser';
 import json from 'rollup-plugin-json';
 import replace from '@rollup/plugin-replace';
+import postcss from 'rollup-plugin-postcss';
```
Comment (spasovski) [inline rollup.config.js:8]: why is this needed?

Comment (openjck) [reply rollup.config.js:8]: I can't say I'm great with Rollup, but I get this error when the postcss plugin is excluded.

```
Error: Unexpected token (Note that you need plugins to import files that are not JavaScript)
```

Comment (hamilton) [reply rollup.config.js:8]: Yes, this is the preferred way to be able to run `import "@graph-paper/core/style.css` and other stylesheets, which is what happens when we run `import { Button } from "@graph-paper/button`.

**Review by spasovski — CHANGES_REQUESTED** [review]
Looks good so far - I like that you updated the stories as well. Only issue is we just introduced the new TempDataForProcessMissingError.svelte which has 4 buttongroups for the probe dimensions. This will need to be rebased and those imports fixed.

File: package.json [hunk 17]
```diff
@@ -45,6 +46,7 @@
   },
   "dependencies": {
     "@auth0/auth0-spa-js": "1.6.0",
+    "@graph-paper/button": "0.0.0-alpha.6",
```
Comment (hamilton) [inline package.json:49]: Can we upgrade this to `alpha.7`? Buttons now take an href as well

**Comment by openjck** [comment]
Closing this so we can get some other stuff in. I'd basically be starting from scratch with a rebase anyway.

**Comment by openjck** [comment]
To be clear, I'll still do this work, but from a different base after we merge some other stuff today.
